bn_s_mp_radix_size_radix_10.c

```diff
@@ -0,0 +1,60 @@
+#include "tommath_private.h"
+#ifdef BN_S_MP_RADIX_SIZE_RADIX_10_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+
+#define LTM_RADIX_SIZE_SCALE 64
+#define LTM_RADIX_SIZE_CONST_SHIFT 16
+int s_mp_radix_size_radix_10(const mp_int *a, int *size)
+{
+   mp_err err;
+   /*
+       floor(2^64/log_2(10)) = 5553023288523357132
+
+       We need that much precision. Example:
+
+         log_2(x) = log(x)/log(2)
+         198096465/log_2(10) = 59632978.000000002598316594477929217520
+
+        Which are more than 16 decimal digits, so with BINARY_64 (C's "double")
+        a 'ceil(198096465/log_2(10))' would result wrongly in 59632978.0.
+    */
+   /*
+       TODO: This is safe for MP_8BIT, change to
+                const uint32_t inv_log_2_10[2] = {0x4d104d42U, 0x7de7fbccU}
+             or branch even further with
+                const uint64_t inv_log_2_10 = {0x4d104d427de7fbccULL};
+             when MP_8BIT got the boot.
+    */
+   const uint16_t inv_log_2_10[4] = {0x4d10u, 0x4d42u, 0x7de7u, 0xfbccu};
```

Comment:
Yes, that would be simpler (and most likely even faster) but I didn't want to add a dependency on a function that you might not use elsewhere and once MP_8BIT is gone it is down to just two `mp_get_u32`, one big-shift and and one big-add.

But if you like it more with `mp_from_ubin`: drop me a note and I'll change it, no problem.

Reply:
Added a branch for `MP_16BIT` which is the only one left, since `MP_8BIT` got the boot, to need that part. Every other size must support 64 bit integers to function, which allow for a simple `mp_set_u64` for that rest.

Reply:
Ok, but please unroll the loop. set, mul, set, add.